app/models/concerns/foreman_templates/ptable_import.rb

```diff
@@ -38,11 +47,22 @@ def import!(name, text, metadata)
     end
 
     # TODO: DRY this, it's copied from ProvisioningTemplateImport
-    def map_oses(metadata)
```

Comment:
I like the refactoring on this method - but we should probably handle this TODO as well. Could you move this method to the TemplateImporter baseclass concern? It should then be available to both of these subclasses.